controllers/goharbor/core/services.go

```diff
@@ -38,3 +39,11 @@ func (r *Reconciler) GetService(ctx context.Context, core *goharborv1alpha2.Core
 		},
 	}, nil
 }
+
+func GetLBService(svc *corev1.Service) *corev1.Service {
```

Comment:
Why do you introduce a new get service method? It seems you should do some modification to the original `reconciler.GetService()` method. The type of the getting service now is determined by the spec data, not directly use the default `ClusterIP` type.